decode.go

```diff
@@ -253,6 +253,16 @@ func (de *decoder) putvalue(wiretype int, val reflect.Value,
 
 	// Embedded message
 	case reflect.Struct:
+		bm := reflect.TypeOf((*encoding.BinaryUnmarshaler)(nil)).Elem()
```

Comment:
Ok it's not possible so do as you like then .. ;) (I guess @bford will have to look at this one for accepting the PR or is it only for `crypto` ?)